### ISSUE TITLE
solve_undetermined_coefficients hangs

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#10169]
Thisexample can't be solved, method hangs
`-8*A - x**5*(A + B + C + E) - x**4*(4*A - 2**(3/4)*C + 4*C + D + 2**(3/4)*E + 4*E + F) - x**3*(-4*2**(3/4)*C + sqrt(2)*C - 2**(3/4)*D + 4*D + sqrt(2)*E + 4*2**(3/4)*E + 2**(3/4)*F + 4*F) - x**2*(4*sqrt(2)*C - 4*2**(3/4)*D + sqrt(2)*D + 4*sqrt(2)*E + sqrt(2)*F + 4*2**(3/4)*F) - x*(2*A + 2*B + 4*sqrt(2)*D + 4*sqrt(2)*F) + 5`

**Comment by smichr** [comment]
in master:
```python
>>> solve_undetermined_coeffs(-8*A - x**5*(A + B + C + E) - x**4*(4*A - 2**(3/4)*C + 4*C + D + 2**(3/4)*E + 4*E + F) - x**3*(-4*2**(3/4)*C + sqrt(2)*C - 2**(3/4)*D + 4*D + sqrt(2)*E + 4*2**(3/4)*E + 2**(3/4)*F + 4*F) - x**2*(4*sqrt(2)*C - 4*2**(3/4)*D + sqrt(2)*D + 4*sqrt(2)*E + sqrt(2)*F + 4*2**(3/4)*F) - x*(2*A + 2*B + 4*sqrt(2)*D + 4*sqrt(2)*F) + 5, var('A:F'),x)
{A: 0.625000000000000, B: -0.00484496124031008, C: -0.369501625625275, D: -0.393418682336328, E: -0.250653413134415, F: 0.174160765689337}
```
There were some issues with the linear solver that probably fixed this issue, too.

**Comment by smichr** [comment]
Let's add a test -- I am not sure which PR fixed this.

**Comment by oscarbenjamin** [comment]
It's not clear to me exactly what the code should be for the original example that failed.

**Comment by oscarbenjamin** [comment]
Looks like this was fixed by 97129c5f70b32ca39875caccf3d5bb574c497546 from #20274
